lib/paper_trail/model_config.rb

```diff
@@ -141,17 +149,24 @@ def cannot_record_after_destroy?
     def setup_versions_association(klass)
       klass.has_many(
         klass.versions_association_name,
-        lambda do |object|
+        lambda do
           relation = order(model.timestamp_sort_order)
-          item_type = object.paper_trail.versions_association_item_type
-          unless item_type.nil?
-            relation = relation.unscope(where: :item_type).where(item_type: item_type)
-          end
+          item_type = klass.paper_trail.send(:versions_association_item_type)
```

Comment:
it's now a public method, so `send` isn't required here

Reply:
Looks like the `.unscope` / `.where` can simply happen `if klass.descendants.empty?`